### DISCUSSION TITLE
Get first name and last name?

### DISCUSSION THREAD

**Discussion by kingmoge** [lupidan/apple-signin-unity#136] [Q&A] [answered]
I see where the plugin returns full name, and seems to maybe guess at the first name (PersonNameFormatterStyle.Short). I know Apple collects First Name and Last Name, and that info seems to be available but I'm unclear if it's available through this plugin, and if it is, I don't get how to access it...

https://developer.apple.com/documentation/authenticationservices/asauthorizationappleidcredential/3180384-fullname/
which leads to...
https://developer.apple.com/documentation/foundation/personnamecomponents

Any ideas?
Thanks!

**Comment by lupidan** [comment, marked as answer]
It´s in the returned Apple ID credential on the first login

https://github.com/lupidan/apple-signin-unity/blob/e39981bca07f08f068231f651e1e8d0f69397222/AppleAuth/Interfaces/IAppleIDCredential.cs#L30

https://github.com/lupidan/apple-signin-unity/blob/master/AppleAuth/Interfaces/IPersonName.cs

**Reply by kingmoge** [reply]
Thanks for the help!

So to make sure I'm reading this right - I would access...
appleIdCredential.FullName.GivenName
appleIdCredential.FullName.FamilyName

Correct?

**Reply by lupidan** [reply]
if you want specific fields, I guess so... That `IPersonName` is a 1:1 on the received `NSPersonNameComponents`. Those fields could be there or not, depends on the user, where they live, their culture...
You also have the option to use the native NSPersonName formatter through the extension method for `IPersonName`, you can specify the style and whether to use the phonetic representation or not.

https://github.com/lupidan/apple-signin-unity/blob/e39981bca07f08f068231f651e1e8d0f69397222/AppleAuth/Extensions/PersonNameExtensions.cs#L12
